var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -182,6 +198,8 @@ def configure_args(self):
         spec = self.spec
         config_args = [
             '--enable-shared',
+            '--with-hwloc-prefix={0}'.format(
+                spec['hwloc'].prefix if '+hwloc' else 'embedded'),
```

Comment:
```suggestion
                spec['hwloc'].prefix if '^hwloc' in spec else 'embedded'),
```
Make sure to only use hwloc if we have actually declared a dependency on it (due to the `depends_on` change above).